File: pm/src/main/java/org/sqlbuilder/pm/VnRoleResolver.java
package org.sqlbuilder.pm;

import org.sqlbuilder.common.DeSerialize;
import org.sqlbuilder.common.Resolver;
import org.sqlbuilder2.ser.Pair;
import org.sqlbuilder2.ser.Triplet;

import java.io.File;
import java.io.IOException;

public class VnRoleResolver extends Resolver<Pair<String,String>, Triplet<Integer,Integer,Integer>>
{
	public VnRoleResolver(final String ser) throws IOException, ClassNotFoundException
	{
		super(DeSerialize.deserialize(new File(ser)));
	}
}

File: pm/src/main/java/org/sqlbuilder/pm/VnRoleResolver.kt
package org.sqlbuilder.pm

import org.sqlbuilder.common.DeSerialize.deserialize
import org.sqlbuilder.common.Resolver
import java.io.File

typealias PmVnRoleResolvable = Pair<String, String>
typealias PmVnRoleResolved = Triple<Int, Int, Int>

class VnRoleResolver(ser: String) : Resolver<PmVnRoleResolvable, PmVnRoleResolved>(deserialize<Map<PmVnRoleResolvable, PmVnRoleResolved>>(File(ser)))
